components/VM_Arm/src/main.c

```diff
@@ -94,7 +95,8 @@ simple_t _simple;
 vspace_t _vspace;
 sel4utils_alloc_data_t _alloc_data;
 allocman_t *allocman;
-static char allocator_mempool[83886080];
+#define ALLOCATOR_MEMPOOL_SZ ((size_t)80 * 1024 * 1024) /* 80 MB*/
+static char allocator_mempool[ALLOCATOR_MEMPOOL_SZ];
```

Comment:
```suggestion
static char allocator_mempool[80*1024*1024];
```
and then using `sizeof ()` avoids the need to have an explicit define here. Seems we do not really need it at them moment and the compiler can handle this well. We could add the define once we make this configurable via CMake.

Reply:
This mempool is tricky. If its not big enough, you can't allocate large amounts of RAM to a guest. That leads to a problem where you have two VMs, one with a large ram need, and one with a small ram need, and you're wasting resources by needing the mempool to be large enough to the high-ram-usage VM.

Reply:
This should really be configurable from the CAmkES component configuration.

Reply:
If someone makes a PR for that, assign me as a reviewer. I did some testing and got a formula for the allocator_mempool -> ram_size relationship.

Reply:
Took care of it myself: https://github.com/seL4/camkes-vm/pull/45/

Reply:
This PR does not affect this behaviour, but shouldn't this pool attempt to align itself more strongly somehow? [allocman seems to deal with unaligned pools](https://github.com/seL4/seL4_libs/blob/0aa8d43fe53564e40248215d2b3b5cedddcc1ee3/libsel4allocman/src/bootstrap.c#L86)¹, but this component should probably take steps to make allocman’s fixup a no-op. This is likely not visible under GCC, which will probably 4K-align this array anyway. Maybe you want `__attribute__((aligned(4096)))`?

¹ Though `sizeof(size_t)` doesn’t seem like the right thing to use here.

Reply:
Does the mspace_pool require a 4k alignment?  I didn't think it did, but if it does then we should reject unaligned init args.

Reply:
> Took care of it myself: #45

That looks nice, I think we should pick this up.
@JoonasOnatsu: could you drop the array changes from this PR, so we have no merge conflicts?

Reply:
> > Took care of it myself: #45
> 
> That looks nice, I think we should pick this up. @JoonasOnatsu: could you drop the array changes from this PR, so we have no merge conflicts?

Will do, that other solution seems much nicer anyways.

Reply:
I dropped the change on the latest push.